vllm/utils.py

```diff
@@ -177,6 +177,32 @@ def clear(self):
         self.cache.clear()
 
 
+class TensorCache:
+    """TODO: Add doc
+    """
+
+    def __init__(self):
+        self._dummy_tensor: torch.Tensor = torch.empty(0)
+        self._tensors: Dict[str, List] = {}
+
+    def get_dummy_tensor(self):
+        return self._dummy_tensor
+
+    def get_tensor(self, name) -> Optional[List]:
+        if name not in self._tensors:
+            return [None, None]
+
+        return self._tensors[name]
+
+    def cache_tensor(self, name: str, tensor_vals: torch.Tensor,
+                     vals: Any) -> None:
+        self._tensors[name] = [tensor_vals, vals]
+
+    def update_vals(self, name: str, vals: Any) -> None:
+        assert name in self._tensors
+        self._tensors[name][1] = vals
+
+
```

Comment:
I find it difficult to understand this class. Can you add some unit tests to make the usecases clear?

Reply:
will do